### DISCUSSION TITLE
Some questions about Menter's SST model on SU2 (might be the bugs)

### DISCUSSION THREAD

**Discussion by sun5k** [su2code/SU2#1483] [Q&A]
Question about Menter's SST model on SU2

Hello! This post is my first Q&A for SU2; happy new Year everyone.
I have two questions about the SST model for the SU2.

The abbreviation is referred from the [SST Turbulence model of NASA TMR](https://turbmodels.larc.nasa.gov/sst.html), [Compressible RANS of NASA TMR](https://turbmodels.larc.nasa.gov/sst.html) , and [Running cases with CFD of NASA TMR](https://turbmodels.larc.nasa.gov/noteonrunning.html). 

Briefly 
1. In SU2, The production term of the SST model uses incompressible assumptions. Is there some special reason? 

2. Always ignore the turbulence kinetic energy to compute the stress tensor for the momentum and energy equation, is it a bug?

Here is the detail!

1.  

The Boussinesq relation of RANS is  

![image](https://user-images.githubusercontent.com/50565314/148731113-5d141cd0-bea0-4bc4-8f2a-e40932645844.png)

Form [Running cases with CFD of NASA TMR](https://turbmodels.larc.nasa.gov/noteonrunning.html): "In the equation above, the![image](https://user-images.githubusercontent.com/50565314/148731540-ace2cac5-4168-48d8-9125-c053806ba463.png) term is sometimes ignored for non-supersonic speed flows, and the **second term in parentheses is identically zero for incompressible flows**."

The production term of SST model is :

![image](https://user-images.githubusercontent.com/50565314/148771167-b96d6d8f-b638-4efd-a6a3-6731ddb4f4a0.png)

Rewrite the first term of the above equation :

![image](https://user-images.githubusercontent.com/50565314/148731731-23cdd717-28d1-453c-a867-77c99494f86f.png)

![image](https://user-images.githubusercontent.com/50565314/148771466-f1613617-0502-4809-9861-a64f4ce0dee7.png)

Finally, the production term becomes :
 
![image](https://user-images.githubusercontent.com/50565314/148732458-c5d7bbd7-bd68-4416-bac2-979aa006512e.png)

This term( ![image](https://user-images.githubusercontent.com/50565314/148732925-d66edf19-484a-4c4e-b518-3cff8ef935f0.png) ) becomes zero in incompressible flow(from continuity equation ![image](https://user-images.githubusercontent.com/50565314/148760571-f54c54eb-a897-4786-a859-c40aee873792.png) ), but the compressible flow is not zero. 

In the SU2 source code( line 849 in [SU2_CFD/src/numerics/turbulent/turb_sources.cpp](https://github.com/su2code/SU2/blob/master/SU2_CFD/src/numerics/turbulent/turb_sources.cpp)), the production term of SU2's SST model is 
"pk = Eddy_Viscosity_i*StrainMag_i*StrainMag_i - 2.0/3.0*Density_i*ScalarVar_i[0]*diverg;" 

Rewrite above code to equation : 
![image](https://user-images.githubusercontent.com/50565314/148732626-7160aca1-11be-4754-8e7d-cb3a8af7dfa5.png)

The omit of  ![image](https://user-images.githubusercontent.com/50565314/148760652-ba17b0d5-c6fe-4169-81e5-8797b1d67dc1.png)  term is a bug, or has some special reason?

![image](https://user-images.githubusercontent.com/50565314/148732925-d66edf19-484a-4c4e-b518-3cff8ef935f0.png) term is not an issue in incompressible flow like low Mach number simulation.  But, in a high Mach number simulation, it may make a noticeable difference.

Q.2)

The stress tensor in RANS simulation is following : 

![image](https://user-images.githubusercontent.com/50565314/148759208-effb0643-a903-48d4-9ef2-3b46162ad5f8.png)

From [Running cases with CFD of NASA TMR](https://turbmodels.larc.nasa.gov/noteonrunning.html)  : "In the equation above, the![image](https://user-images.githubusercontent.com/50565314/148731540-ace2cac5-4168-48d8-9125-c053806ba463.png) term is **sometimes ignored for non-supersonic speed flows**, and the second term in parentheses is identically zero for incompressible flows." And, from [SST Turbulence model of NASA TMR](https://turbmodels.larc.nasa.gov/sst.html) : "**When the ![image](https://user-images.githubusercontent.com/50565314/148762159-e2b08305-e205-4ffd-9ec1-e9de19cda42d.png) term is ignored in ![image](https://user-images.githubusercontent.com/50565314/148762216-7e565ca3-856d-4c18-8e16-c9ad780452b8.png) in the momentum and energy equations** and the production term is approximated by ![image](https://user-images.githubusercontent.com/50565314/148762331-d102ea3b-b5fe-42d3-aa83-ffa0f18e3104.png), a modified naming convention should be employed : **SSTm**"

In the SU2 source code( line 142 in [SU2_CFD/src/numerics/flow/flow_diffusion.cpp](https://github.com/su2code/SU2/blob/master/SU2_CFD/src/numerics/flow/flow_diffusion.cpp)), The turbulence kinetic energy of the stress tensor is always calculated as 0. I don't know who, someone comments as 
"// TODO why ignore turb_ke?"
This comment existed until SU2 v.7.1.0 to SU2 v.7.2.1(I'm not sure the same about the bellow SU2 v710).

I guess the SST model of SU2 is 'SST' not 'SSTm'. Then, it should be used ![image](https://user-images.githubusercontent.com/50565314/148767053-79a012e3-1061-46b9-b432-359b7b04969f.png) in the stress tensor of the momentum and energy equations.

The code always ignores the turbulence kinetic energy to compute the stress tensor for the momentum and energy equation.
 Is it under maintenance? 

Extra question) Someone working to implement for other kinds of SST(like SST-V, SST-Vm, SST-2003, SST-2003m, ... )? 

Thank you for reading.

**Comment by bigfooted** [comment]
Hi! 
Yes, these divergence terms are often neglected, even in commercial codes like ANSYS Fluent they are absent in the production. Nevertheless, when we say that the model is "SST 1994" or whatever, then the model should be what is reported. So I am supporting that we carefully check the differences with the literature model and implement it exactly as reported. When we find convergence issues, we can implement different versions of SST from literature or use a config option to switch a term on or off, but at least then the user knows exactly what is being solved.  
Note that in a more recent version of SST, this divergence term is neglected and the strain is replaced by the vorticity.

**Reply by sun5k** [reply]
Thank you for your reply. 

I really agree with your opinion that CFD users must know what they do.

**Comment by pcarruscag** [comment]
Hi folks, just leaving a link here to #1364 with a related discussion.

**Reply by TobiKattmann** [reply]
https://github.com/su2code/SU2/discussions/1403 might be interesting as well

**Comment by pcarruscag** [comment]
And another more pertinent to SST here #797

**Comment by pcarruscag** [comment]
@sun5k Do you want to try making the model consistent with SSTm? Based on #797 we want SSTm instead of SST because the results near the wall are much better. But we don't seem to have switched completely from one to the other as you have found.

**Reply by sun5k** [reply]
I think a clear name is best. it makes less confusion. I agree with your suggestion of #1551.

Thank you for the mention.

**Comment by WallyMaier** [comment]
Hi @sun5k,
I haven't been following the github very closely recently, but there is a good chance I will at least attempt to implement SST-Vm.   I dont think it should be too much work, I just need to find the time.....
